scripts/benchmark/utils/decorator.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import os
+import timeit
+
+from typing import (
+    Any,
+    Callable
+)
+
+
+def default_format(val: str) -> str:
+    return val
+
+
+# TODO: This is a rip off from profilehooks. Clean this up
+
+def timecall(
+        fn: Callable[..., Any]=None,
+        log_level: int=logging.INFO,
+        formatter: Callable[[str], str]=default_format) -> Callable[..., Any]:
+
+    if fn is None:  # @timecall() syntax -- we are a decorator maker
+        def decorator(fn: Callable[..., Any]) -> Callable[..., Any]:
+            return timecall(fn, log_level=log_level, formatter=formatter)
+        return decorator
+
+    # @timecall syntax -- we are a decorator.
+    fp = FuncTimer(fn, log_level=log_level, formatter=formatter)
+
+    # We cannot return fp or fp.__call__ directly as that would break method
+    # definitions, instead we need to return a plain function.
+    def new_fn(*args: Any, **kw: Any) -> Any:
+        return fp(*args, **kw)
+    new_fn.__doc__ = fn.__doc__
```

Comment:
Instead of all of this I think using `functools.wraps` is the proper utility to decorate the `new_fn`

Reply:
Thanks for the pointer. I had the feeling this could be improved. This file is largely copied from somewhere :)